rfc/2019-12-07-fuzz-and-auto-debug.md

```diff
@@ -0,0 +1,161 @@
+   |           |mutation|       |                                                                      |
+   |           +---+----+       |                                                                      |
+   | mutated input |            |                                                                      |
+   |               v            |                                                                      |
+   |           +---+-----+      |     +---------------+                                                |
+   |           |execution+----------->+  wrapped app  |                                                |
+   |           +---+-----+      |     +---+-----------+                                                |
+   |  exec result  |            |         ^                                                            |
+   |               v            |         |                                                            |
+   |            +--+---+        |         | get trace info of this input                               |
+   |            |check |        |         |                                                            |
+   |            +--+---+        |         |                                                            |
+   |               |            |         |                                                            |
+   +----------------------------+         |                                                            |
+                   |            +-----------------------------------------------+                      |
+                   |            |         |                        Debug module |                      |
+           input,  |            |         |                                     |                      |
+check pass or not  |            |         |                                     |                      |
+                   |            | +-------+-+       +-----------------+         |                      |
+                   +------------> |collector+------>+statistical model+--------------------------------+
+                                | +---------+       +-------+---------+         |
+                                |                           ^                   |
+                                |                           |                   |
+                                +-----------------------------------------------+
+                                                            |
+                                                            |
+                                                       +----+---+
+                                                       |frontend|
+                                                       +--------+
+
+```
+
+
+
+### Instruction tool
+
+
+
+Like [go tool cover](), we will add code at the beginning of every code basic block in conpiler phase:
+
+```go
+        trace_util_0.Count(_main_00000, 0)   // Instruction
```

Comment:
please remove the prefix spaces.

Reply:
OK, I have fixed it

Reply:
The prefix spaces still exists, I'm going to reopen this conversion.

Reply:
Do you mean prefix spaces in code? I mistake it to the empty line behind the code block...